### PR TITLE
Label positioning fix

### PR DESCRIPTION
**Pull request by alburdette619**
Fixed: 

- #206 the focused/filled label positioning by basing the size of the text off a font size animation instead of a scale.  The scale was causing us to have to calculate exactly how far each label based on width needed to be translated, but this seems to just work.  Also changed some of the positioning to better fit material, please check me there.

- The label was wrapped in a View such that pointer events can be ignored.  Previously, this was causing taps to be ignored on the label text and focus would not occur.

- The labelStyle passed in from the TextField sub class was not being applied.

### CODE REVIEW

**Comment by alburdette619** [comment]
I'll get this build fixed. Looks like it's linting and snapshots now.

**Comment by codypearce** [comment]
@alburdette619 Looks good, but something isn't being calculated correctly on Outlined TextFields now

image

**Comment by alburdette619** [comment]
This should be g2g now @codypearce.  I had to account for the animation stops being different on web vs native.

**Comment by codypearce** [comment]
@alburdette619 Look good, last thing: the label on the outlined textfield needs to have a padding on either side to create the illusion of breaking the outline.

You can see this in the docs https://material.io/design/components/text-fields.html


image

File: src/Components/TextField/TextFieldLabel/TextFieldLabel.js [hunk 1]
```diff
@@ -163,24 +187,44 @@ class TextFieldLabel extends Component {
...
     return (
-      <Animated.Text
+      <Animated.View
         style={[
-          styles.label,
+          styles.container,
           {
-            color: labelColor,
-            backgroundColor: type == 'outlined' ? 'white' : 'transparent',
-            paddingHorizontal: type == 'outlined' ? 4 : 0,
```
Comment (codypearce) [inline src/Components/TextField/TextFieldLabel/TextFieldLabel.js]: This is the line should be added back

**Comment by codypearce** [comment]
The labels for dense should be smaller
Components | TextField : Outlined - dense ⋅ Storybook 2019-06-21 20-47-07

as in here https://codypearce.github.io/material-bread/?path=/story/components-textfield-outlined--dense

**Comment by codypearce** [comment]
Components | TextField : Filled - with prefix ⋅ Storybook 2019-06-21 20-48-43
Components | TextField : Outlined - with prefix ⋅ Storybook 2019-06-21 20-49-32
Components | TextField : Flat - with prefix ⋅ Storybook 2019-06-21 20-49-22

The prefix labels are too large as well

**Comment by codypearce** [comment]
I think the label position on the top got moved over as well 
Components | TextField : Outlined - simple ⋅ Storybook 2019-06-21 20-51-34

**Comment by alburdette619** [comment]
@codypearce lemme know if there's anything else.  I tried to follow material to the specs exactly on these.  I fixed a couple of small issues I found along the way.

**Comment by alburdette619** [comment]
Also, the font animation looked good with the easing on RN, but on the web it looked odd, so I made that OS specific.

**Comment by codypearce** [comment]
@alburdette619 Wow this looks really good, animation is smooth and other fixes work great!

Only thing I see is that after the label is animated, it needs to be smaller

Text fields - Material Design 2019-06-23 23-54-00


Components | TextField : Flat - simple ⋅ Storybook 2019-06-23 23-56-11

**Comment by alburdette619** [comment]
I'll check, but I'm already making them smaller than .75 of the original font size which is smaller than is shown here.

**Comment by alburdette619** [comment]
oic, there is a bug causing it to not get smaller, got it.

**Comment by alburdette619** [comment]
@codypearce okay, fixed that bug.  Turned out to be two different bugs (style order, and null easing).  Should be good on native & web this time 🤞

**Comment by codypearce** [comment]
@alburdette619 Everything looks good, thanks!

**Pull request merged**